### DISCUSSION TITLE
statusbar styles decision

### DISCUSSION THREAD

**Discussion by jcesarmobile** [apache/cordova#372] [Polls]
The statusbar plugin has this styles:
StatusBar.styleDefault: Use the default statusbar (dark text, for light backgrounds)
StatusBar.styleLightContent: Use the lightContent statusbar (light text, for dark backgrounds).
This was enough until iOS 13 came out, then Apple added DarkContent to be the dark text and changed default to change according to the user theme.

Since StatusBar.styleDefault was setting the text to whie in some cases now, I changed it to work as the docs say, always use dark text (asked on the issue and users wanted it like that). But after it was merged, some users started to say that we should provide a new StatusBar.styleDarkContent and make StatusBar.styleDefault work as the native default value (change according to the configured user theme).

So, what should we do?

1) Add a new StatusBar.styleDarkContent method that would work as the StatusBar.styleDefault works now, dark text, and change StatusBar.styleDefault to use the native default value so it changes according to the user theme.

2) Add a new StatusBar.styleDarkContent method that would work as the styleDefault works now, dark text, add another new StatusBar.styleSystem (or similar name)  to use the native default value so it changes according to the user theme, and deprecate StatusBar.styleDefault.

3) Keep the StatusBar.styleDefault for dark text and add a new StatusBar.styleSystem (or similar name) to make the status bar change according to the user theme.

Next release is going to be a major release, so we shouldn't worry about breaking changes. But not sure which change would be the least confusing for users, matching the style names to the native iOS names and their bebavior, keep default as dark text as that's what is documented or use a more neutral name for style changing according to the theme.

Note, it's not clear if we can accomplish the status bar style change according to the user/phone theme on Android.

**Comment by dpogue** [comment]
In a perfect world, the status bar background would either be provided as part of the page (when using `viewport-fit=cover`) or as a solid colour from the meta `theme-color` tag with the text colour automatically set based on contrast requirements.

Unfortunately, the API to read the `theme-color` value wasn't added to WKWebView until iOS 15, and no such API exists on Android to read either the `theme-color` or the `viewport-fit` value 😞

**Reply by jcesarmobile** [reply]
If we wanted that behaviour we could probably use javascript to get the `theme-color` value from the `index.html` for the iOS/Android versions where there is no API for getting the value natively. There is also an `apple-mobile-web-app-status-bar-style` meta tag we could try to read if we wanted users to set the values from the `index.html` instead of doing it from `config.xml`. (I think this meta tag it's deprecated)
But I think that's a separate discussion.
